botocore/docs/utils.py

```diff
@@ -96,6 +96,33 @@ def __new__(cls, name, type_name, documentation, metadata=None,
             required_members)
 
 
+class AutoPopulatedParam(object):
+    def __init__(self, name):
+        self.name = name
+
+    def document_auto_populated_param(self, event_name, section, **kwargs):
+        """This documents the auto populated parameters
+
+        It will remove any required marks for the parameter, remove the
+        parameter from the example, and add a snippet about the parameter
+        being autopopulated in the description.
+        """
+        if event_name.startswith('docs.request-params'):
+            if self.name in section.available_sections:
+                section = section.get_section(self.name)
+                if 'is-required' in section.available_sections:
+                    section.delete_section('is-required')
+                description_section = section.get_section(
+                    'param-documentation')
+                description_section.writeln(
+                    'Note this parameter is autopopulated. There is no '
```

Comment:
What about moving this up to a class constant and allowing a custom message to be passed in at instantiation?  I was thinking that for some of the other handlers that will use this, we might want to provide a slightly different doc message.